src/psyclone/transformations.py

```diff
@@ -4368,3 +4368,88 @@ def validate(self, node_list, options=None):
+    the values/references are passe by argument into the kernel.'''
+
+    @property
+    def name(self):
+        ''' Returns the name of this transformation as a string.'''
+        return "KernelGlobalsToArguments"
+
+    def __str__(self):
+        return ("Convert the global variables used inside the kernel "
+                "into arguments and modify the InvokeSchedule to pass them"
+                " in the kernel call.")
+
+    def validate(self, node, options=None):
+        '''
+        Check that the supplied node is a valid target for this transformation.
+
+        :param node: the loop node to validate.
+        :type node: :py:class:`psyclone.psyGen.CodedKern`
+        :param options: a dictionary with options for transformations.
+        :type options: dictionary of string:values or None
+
+        :raises TransformationError: if the supplied node is not a CodedKern
+        '''
+        from psyclone.psyGen import CodedKern
+        if not isinstance(node, CodedKern):
+            raise TransformationError(
+                "The KernelGlobalsToArguments transformation can "
+                "only be applied to CodedKern nodes but found {0} instead."
+                "".format(type(node)))
+
+    def apply(self, node, options=None):
+        '''
+        Convert the global variables used inside the kernel into arguments and
+        modify the InvokeSchedule to pass the same global variables to the
+        kernel call.
+
+        :param node: a kernel call
+        :type nodes: :py:class:`psyclone.psyGen.CodedKern`
+        :param options: a dictionary with options for transformations.
+        :type options: dictionary of string:values or None
+
+        :returns: tuple of the modified Schedule and a record of the \
+                  transformation.
+        :rtype: (:py:class:`psyclone.psyGen.Schedule`, \
+                 :py:class:`psyclone.undoredo.Memento`).
+        '''
+        from psyclone.psyir.symbols import ArgumentInterface
+
+        self.validate(node)
+
+        kernel = node.get_kernel_schedule()
+        symtab = kernel.symbol_table
+        invoke_symtab = node.root.symbol_table
+
+        # Transform each global variable into an argument
+        for globalvar in kernel.symbol_table.global_datasymbols:
+
+            # Resolve the data type information if it is not available
+            if globalvar.datatype == 'deferred':
+                globalvar.resolve_deferred()
+
+            # Copy the global into the InvokeSchedule SymbolTable
+            invoke_symtab.copy_external_global(globalvar)
+
+            # Convert the symbol to an argument and add it to the argument list
+            current_arg_list = symtab.argument_list
+            globalvar.interface = ArgumentInterface(
+                ArgumentInterface.Access.READWRITE)
+            current_arg_list.append(globalvar)
+            symtab.specify_argument_list(current_arg_list)
+
+            # Add the Global variable in the call argument list
```

Comment:
This section feels like it belongs in a new method e.g. `node.arguments.add(type, name, access)`, i.e. create a new method in the psyGen.Arguments base class. Is that do-able?

Reply:
Done, but I used `append` as method name as I just used it to add at the end of the argument list.